### ISSUE TITLE
Does Mitogen need python-distro? Fails in facts gathering if python-distro is not installed.

### ISSUE THREAD

**Issue by ElCoyote27** [dw/mitogen#610]
Hi,
I'm using mitogen HEAD (as of 2019/08/06) on a RHEL7.6 host with ansible 2.8.
I've noticed that RHEL7/RHEL8 and Debian9 hosts fail to gather facts if 'python-distro' is not installed.
E.g:
```TASK [Gathering Facts] ****************************************************************************************************************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was:   File "<stdin>", line 1414, in load_module
fatal: [dc02.lasthome.solace.krynn]: FAILED! => {"msg": "Unexpected failure during module execution.", "stdout": ""}
```

**Comment by ElCoyote27** [comment]
running with -vvv gives the following backtrace:
```
TASK [Gathering Facts] ****************************************************************************************************************************************************************
task path: /export/home/raistlin/World/Vincent/Code/GIT/krynn-ansible/all.yml:3
The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 145, in run
    res = self._execute()
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 650, in _execute
    result = self._handler.run(task_vars=variables)
  File "/export/home/raistlin/World/Vincent/Code/GIT/krynn-ansible/mitogen/ansible_mitogen/mixins.py", line 116, in run
    return super(ActionModuleMixin, self).run(tmp, task_vars)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/gather_facts.py", line 66, in run
    res = self._execute_module(module_name=fact_module, module_args=mod_args, task_vars=task_vars, wrap_async=False)
  File "/export/home/raistlin/World/Vincent/Code/GIT/krynn-ansible/mitogen/ansible_mitogen/mixins.py", line 346, in _execute_module
    self._temp_file_gibberish(module_args, wrap_async)
  File "/export/home/raistlin/World/Vincent/Code/GIT/krynn-ansible/mitogen/ansible_mitogen/mixins.py", line 325, in _temp_file_gibberish
    self._connection.get_good_temp_dir()
  File "/export/home/raistlin/World/Vincent/Code/GIT/krynn-ansible/mitogen/ansible_mitogen/connection.py", line 762, in get_good_temp_dir
    self._connect()
  File "/export/home/raistlin/World/Vincent/Code/GIT/krynn-ansible/mitogen/ansible_mitogen/connection.py", line 782, in _connect
    self._connect_stack(stack)
  File "/export/home/raistlin/World/Vincent/Code/GIT/krynn-ansible/mitogen/ansible_mitogen/connection.py", line 735, in _connect_stack
    stack=mitogen.utils.cast(list(stack)),
  File "/export/home/raistlin/World/Vincent/Code/GIT/krynn-ansible/mitogen/mitogen/service.py", line 113, in call
    return call_context.call_service(service_name, method_name, **kwargs)
  File "/export/home/raistlin/World/Vincent/Code/GIT/krynn-ansible/mitogen/mitogen/core.py", line 2086, in call_service
    return recv.get().unpickle()
  File "/export/home/raistlin/World/Vincent/Code/GIT/krynn-ansible/mitogen/mitogen/core.py", line 916, in unpickle
    raise obj
CallError: exceptions.ImportError: The Mitogen master process was unable to serve u'ansible.module_utils.distro._distro'. It may be a native Python extension, or it may be missing entirely. Check the importer debug logs on the master for more information.
  File "<stdin>", line 3354, in _dispatch_one
  File "<stdin>", line 3341, in _parse_request
  File "<stdin>", line 622, in import_module
  File "<stdin>", line 1441, in load_module
  File "<string>", line 1, in <module>
[...]
```

**Comment by ElCoyote27** [comment]
I was able to solve it with:
'yum install python-distro' (on RHEL7)
'yum install python2-distro python3-distro' (on RHEL8)
'apt-get install python-distro' (Debian 9)

**Comment by dw** [comment]
Thanks for reporting. This is a duplicate of #590

You should not need to install python-distro. Ansible is corrupting `sys.modules` in 2.8 and it confuses Mitogen's importer. Please subscribe to #590 to be notified when a fix is available.

**Comment by ElCoyote27** [comment]
Thanks for the quick reply. I'll subscribe to the other issue.

**Comment by stefanor** [comment]
@dw: It seems #590 got moved to a private repo, which really messes with tracking this issue :(

Seems the commit that fixed 590 was 8f940e2ccb7df4aafdd9ecf815fceca3aef40ec2

**Comment by stefanor** [comment]
From https://bugs.debian.org/1006330:

Something like this bug seems to be back with distro 1.7.0. Presumably because of the change of module layout in python-distro/distro#315
